### DISCUSSION TITLE
Is it possible AT ALL to select a git resource branch interactively in web UI or fly?

### DISCUSSION THREAD

**Discussion by saldoukhov** [concourse/concourse#6463] [Help & Support]
F.e. normally you use "develop", but today you want to use "patchXYZ"... Without changing the pipeline, is there a way to prompt the user for the branch name?

**Comment by taylorsilva** [comment]
There is no way to do this within Concourse. The web UI is really just a read-only interface (for the most part). 

I think the main example of running the same pipeline for different branches/repos is the resources pipelines:
https://ci.concourse-ci.org/?search=team%3A%22main%22%20group%3A%22resource%22
<img width="830" alt="image" src="https://user-images.githubusercontent.com/4583837/105739361-07309900-5f06-11eb-9810-4caa07adb9a3.png">

To keep those pipelines updated we have a job that does `set_pipline` for each pipeline:
<img width="402" alt="image" src="https://user-images.githubusercontent.com/4583837/105739513-2e876600-5f06-11eb-9b31-b41f5c1e91e7.png">

A more concrete example is here: https://github.com/concourse/examples/blob/7208e3f1303d2027b30215fb939b081b89ca6b18/pipelines/set-pipelines.yml#L36-L47
which renders as this pipeline: https://ci.concourse-ci.org/teams/examples/pipelines/set-pipelines
which then creates these pipelines: https://ci.concourse-ci.org/?search=team%3A%22examples%22%20group%3A%22instance-groups%22

There's also this blog post that explains things in more details about doing gitops for your pipelines: https://blog.concourse-ci.org/gitops-for-pipelines/

Hope that helps and gives you some ideas :)
